### DISCUSSION TITLE
Converting 'foo = foo.Trim()' over to 'foo .= Trim()' or something to that effect

### DISCUSSION THREAD

**Discussion by dsidirop** [dotnet/csharplang#4245] [Language Ideas]
Oftentimes we find ourselves trimming strings and what have you so I thought maybe it would make sense to support

foo .= Trim();

Feel free to suggest a more elegant syntax (but you get the idea). What does the community think?

**Comment by Unknown6656** [comment]
I proposed this way back in the roslyn repo: https://github.com/dotnet/roslyn/issues/9134
I'll quote @HaloFour 's reply on using `.=`:

> @Unknown6656
> Or you could assign the next-to-last result from the properties to a temporary variable:
> ``` cs
> MyControl ctrl = .....;
> Control parent = ctrl.Parent.Parent.Parent.Parent.Parent.Parent.Parent;
> parent.ExampleList = parent.ExampleList.Reverse();
> ```

_Originally posted by @HaloFour in https://github.com/dotnet/roslyn/issues/9134#issuecomment-189038461_

It seems that there is not really a need for this kind of operator/syntactic sugar. I have to say, that I do quite agree when looking back.

**Reply by osamakawish** [reply]
I think you closed that thread prematurely. The point of this is to shorten and clean the code. Also, your thread was slightly different and well, you started off using nested classes (which I don't think I have actually even used for quite a while), and I think a lot of comments in that thread apply to nested classes than the notation @dsidirop is suggesting.

**Comment by theunrepentantgeek** [comment]
This exact idea - plus alternatives syntax ideas - has been discussed before, and has always floundered on the question of value. 

Any proposal that has the _sole benefit_ of making C# more concise has to address an extremely common case to be worth the cost of implementation. 

This is why changes like expression based methods were implemented while a slew of equally simple but less commonly applicable proposals have been passed over. 

In my experience, reassigning an existing variable to the result of calling a method on that exact instance is not only rare, but often a source of bugs as well.
